ovh/types_cloud_project_database.go

```diff
@@ -1995,3 +1995,177 @@ func (v CloudProjectDatabaseKafkaUserAccessResponse) ToMap() map[string]interfac
+	DatabaseId *string                                               `json:"databaseId,omitempty"`
+	Mode       *CloudProjectDatabasePostgresqlConnectionPoolModeEnum `json:"mode,omitempty"`
+	Size       *int                                                  `json:"size,omitempty"`
+	UserId     *string                                               `json:"userId"`
+}
+
+func (opts *CloudProjectDatabasePostgresqlConnectionPoolUpdateOpts) FromResource(d *schema.ResourceData) *CloudProjectDatabasePostgresqlConnectionPoolUpdateOpts {
+	opts.DatabaseId = helpers.GetNilStringPointer(d.Get("database_id"))
+	mode := helpers.GetNilStringPointer(d.Get("mode"))
+	if mode != nil {
+		enumMode := CloudProjectDatabasePostgresqlConnectionPoolModeEnum(*mode)
+		opts.Mode = &enumMode
+	}
+	opts.Size = helpers.GetNilIntPointer(d.Get("size"))
+	opts.UserId = helpers.GetNilStringPointer(d.Get("user_id"))
+	return opts
+}
+
+type CloudProjectDatabasePostgresqlConnectionPoolResponse struct {
+	DatabaseId string                                                   `json:"databaseId"`
+	Id         string                                                   `json:"id"`
+	Mode       CloudProjectDatabasePostgresqlConnectionPoolModeEnum     `json:"mode"`
+	Name       string                                                   `json:"name"`
+	Port       int64                                                    `json:"port"`
+	Size       int                                                      `json:"size"`
+	SslMode    *CloudProjectDatabasePostgresqlConnectionPoolSslModeEnum `json:"sslMode,omitempty"`
+	Uri        string                                                   `json:"uri"`
+	UserId     *string                                                  `json:"userId,omitempty"`
+}
+
+func (v CloudProjectDatabasePostgresqlConnectionPoolResponse) ToMap() map[string]interface{} {
+	obj := make(map[string]interface{})
+
+	obj["database_id"] = v.DatabaseId
+	obj["id"] = v.Id
+	obj["mode"] = v.Mode
+	obj["name"] = v.Name
+	obj["port"] = v.Port
+	obj["size"] = v.Size
+	obj["ssl_mode"] = v.SslMode
+	obj["uri"] = v.Uri
+	obj["user_id"] = v.UserId
+
+	return obj
+}
+
+func importCloudProjectDatabasePostgresqlConnectionPool(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	n := 3
+	splitId := strings.SplitN(givenId, "/", n)
+	if len(splitId) != n {
+		return nil, fmt.Errorf("Import Id is not service_name/cluster_id/id formatted")
+	}
+	serviceName := splitId[0]
+	clusterId := splitId[1]
+	id := splitId[2]
+	d.SetId(id)
+	d.Set("cluster_id", clusterId)
+	d.Set("service_name", serviceName)
+
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func postCloudProjectDatabasePostgresqlConnectionPool(ctx context.Context, d *schema.ResourceData, meta interface{}, dsReadFunc, readFunc schema.ReadContextFunc, updateFunc schema.UpdateContextFunc, f func() interface{}) diag.Diagnostics {
```

Comment:
I guess that this function is created on the model of the `postCloudProjectDatabaseUser` function but no need here of `dsReadFunc`, `updateFunc` used for a special case and no need the `f` function parameter used to make `postCloudProjectDatabaseUser` genirc because use by different engine with different params.